lib/search/format_migrator.rb

```diff
@@ -1,7 +1,8 @@
 module Search
   class FormatMigrator
-    def initialize(base = nil)
-      @base = base
+    def initialize(base_query: nil, cluster: Clusters.default_cluster)
```

Comment:
Given that all the `cluster` is used for is to get the migrated indices, would be be better to have a `migrated_indices` parameter?  It means `FormatMigrator` doesn't need to know about clusters.  `QueryBuilder` still needs to (for now) but that's at least a little further towards the boundaries of the system.

Reply:
I like the idea. It would mean duplicating `SearchConfig.instance.new_content_index(cluster: cluster).real_index_names` everywhere we use `FormatMigrator` though, and it might not be obvious why.

This way at least we know that FormatMigrator depends on a specific cluster through the index names.

Reply:
Hmm, the duplication would be unfortunate.  Maybe leave it as it is for now then and we can think more later about how to handle this cluster proliferation.  I'm still leaning towards factory methods as the right thing to do.

Reply:
OK. If you have ideas on how to do the factory methods in a nice way I'd be keen to add them!